seastar/src/submit_to.rs

```diff
@@ -0,0 +1,122 @@
+use crate as seastar;
+use ffi::*;
+use std::future::Future;
+
+use crate::cxx_async_local_future::IntoCxxAsyncLocalFuture;
+
+#[cxx::bridge]
+mod ffi {
+    #[namespace = "seastar_ffi"]
+    unsafe extern "C++" {
+        type VoidFuture = crate::cxx_async_futures::VoidFuture;
+    }
+
+    #[namespace = "seastar_ffi::submit_to"]
+    unsafe extern "C++" {
+        include!("seastar/src/submit_to.hh");
+
+        unsafe fn submit_to(
+            shard_id: u32,
+            closure: *mut u8,
+            caller: unsafe fn(*mut u8) -> VoidFuture,
+        ) -> VoidFuture;
+    }
+}
+
+/// Convert the pointer to Box<Func> and then call it, consuming it in the process.
+fn fn_once_caller<Func>(raw_func: *mut u8) -> VoidFuture
+where
+    Func: FnOnce() -> VoidFuture,
```

Comment:
Nit: it could probably be made more generic:

```rust
fn fn_once_caller<Func, Ret>(raw_func: &mut u8) -> VoidFuture
where
    Func: FnOnce() -> Ret
{
    // ...
}
```

You can do something similar to `get_fn_once_caller`.